File: app/src/main/java/com/example/ui/screens/AiCareerScreen.kt
package com.example.ui.screens

import android.content.Context
import android.widget.Toast
import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Clear
import androidx.compose.material.icons.filled.ContentCopy
import androidx.compose.material.icons.outlined.AutoAwesome
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.viewmodel.MainViewModel
import kotlinx.coroutines.delay

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AiCareerScreen(
    viewModel: MainViewModel,
    modifier: Modifier = Modifier
) {
    val aiOutput by viewModel.aiOutput.collectAsState()
    val isLoading by viewModel.isAiLoading.collectAsState()

    val context = LocalContext.current
    val clipboardManager = LocalClipboardManager.current

    // Rotating tips during AI Generation
    val loadingTips = listOf(
        "Synthesizing job requirements with your skills matrix...",
        "Structuring a highly persuasive cover letter...",
        "Identifying vital technical keywords to beat automated ATS...",
        "Formulating realistic, customized interview practice questions...",
        "Polishing credentials with professional recruiter vocabulary..."
    )
    var currentTipIndex by remember { mutableStateOf(0) }

    LaunchedEffect(isLoading) {
        if (isLoading) {
            while (true) {
                delay(3500)
                currentTipIndex = (currentTipIndex + 1) % loadingTips.size
            }
        }
    }

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
    ) {
        // Toolbar / Action header
        Surface(
            modifier = Modifier.fillMaxWidth(),
            tonalElevation = 2.dp,
            shadowElevation = 2.dp,
            color = MaterialTheme.colorScheme.surface
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp, vertical = 8.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Icon(
                        imageVector = Icons.Default.AutoAwesome,
                        contentDescription = "AI",
                        tint = MaterialTheme.colorScheme.primary
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        text = "AI Career Advisor",
                        style = MaterialTheme.typography.titleLarge,
                        fontWeight = FontWeight.Bold
                    )
                }

                if (aiOutput.isNotEmpty() && !isLoading) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        IconButton(
                            onClick = {
                                clipboardManager.setText(AnnotatedString(aiOutput))
                                Toast.makeText(context, "Copied to Clipboard! 📋", Toast.LENGTH_SHORT).show()
                            },
                            modifier = Modifier.testTag("copy_ai_output_btn")
                        ) {
                            Icon(Icons.Default.ContentCopy, "Copy Details")
                        }

                        IconButton(
                            onClick = { viewModel.clearAiOutput() },
                            modifier = Modifier.testTag("clear_ai_output_btn")
                        ) {
                            Icon(Icons.Default.Clear, "Clear Draft")
                        }
                    }
                }
            }
        }

        // Output space
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(16.dp),
            contentAlignment = Alignment.TopCenter
        ) {
            when {
                isLoading -> {
                    Column(
                        modifier = Modifier
                            .fillMaxSize()
                            .padding(top = 80.dp),
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(16.dp)
                    ) {
                        CircularProgressIndicator(
                            color = MaterialTheme.colorScheme.primary,
                            strokeWidth = 4.dp,
                            modifier = Modifier.size(52.dp)
                        )
                        Spacer(modifier = Modifier.height(12.dp))
                        Text(
                            text = "Drafting with Gemini...",
                            style = MaterialTheme.typography.titleMedium,
                            fontWeight = FontWeight.Bold
                        )
                        // Rotating professional advice panel
                        Card(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 16.dp),
                            colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f))
                        ) {
                            Text(
                                text = "💡 ${loadingTips[currentTipIndex]}",
                                style = MaterialTheme.typography.bodyMedium,
                                lineHeight = 20.sp,
                                modifier = Modifier.padding(16.dp),
                                textAlign = TextAlign.Center,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }

                aiOutput.isEmpty() -> {
                    Column(
                        modifier = Modifier
                            .fillMaxSize()
                            .padding(top = 80.dp),
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(16.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Outlined.AutoAwesome,
                            contentDescription = null,
                            modifier = Modifier.size(64.dp),
                            tint = MaterialTheme.colorScheme.outline
                        )
                        Text(
                            text = "AI Advisor Standing By",
                            style = MaterialTheme.typography.titleLarge,
                            fontWeight = FontWeight.Bold
                        )
                        Text(
                            text = "Navigate to any job's detail page and tap one of our AI tools:\n\n• ✍️ Draft Cover Letter\n• 🎯 Tailor Resume\n• 🎤 Prepare Interview\n\nConfigure your Profile first to let the AI draft details tailored specifically to your background!",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            textAlign = TextAlign.Center,
                            modifier = Modifier.padding(horizontal = 24.dp),
                            lineHeight = 22.sp
                        )
                    }
                }

                else -> {
                    // Styled AI response panel
                    Card(
                        modifier = Modifier
                            .fillMaxSize()
                            .verticalScroll(rememberScrollState()),
                        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
                        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp)
                    ) {
                        Column(
                            modifier = Modifier.padding(16.dp)
                        ) {
                            Surface(
                                color = MaterialTheme.colorScheme.primaryContainer,
                                shape = RoundedCornerShape(8.dp),
                                modifier = Modifier.padding(bottom = 16.dp)
                            ) {
                                Row(
                                    verticalAlignment = Alignment.CenterVertically,
                                    modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp)
                                ) {
                                    Icon(
                                        imageVector = Icons.Default.AutoAwesome,
                                        contentDescription = null,
                                        tint = MaterialTheme.colorScheme.primary,
                                        modifier = Modifier.size(16.dp)
                                    )
                                    Spacer(modifier = Modifier.width(6.dp))
                                    Text(
                                        text = "Gemini Powered Draft",
                                        style = MaterialTheme.typography.labelSmall,
                                        fontWeight = FontWeight.Bold,
                                        color = MaterialTheme.colorScheme.onPrimaryContainer
                                    )
                                }
                            }

                            Text(
                                text = aiOutput,
                                style = MaterialTheme.typography.bodyMedium,
                                lineHeight = 24.sp,
                                modifier = Modifier.testTag("ai_output_text"),
                                fontFamily = FontFamily.Default
                            )

                            Spacer(modifier = Modifier.height(48.dp))
                        }
                    }
                }
            }
        }
    }
}

File: lib/screens/ai_career_screen.dart
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:provider/provider.dart';
import '../providers/jobs_provider.dart';

class AiCareerScreen extends StatefulWidget {
  const AiCareerScreen({super.key});

  @override
  State<AiCareerScreen> createState() => _AiCareerScreenState();
}

class _AiCareerScreenState extends State<AiCareerScreen> {
  Timer? _tipTimer;
  int _currentTipIndex = 0;

  final List<String> _loadingTips = [
    'Synthesizing job requirements with your skills matrix...',
    'Structuring a highly persuasive cover letter...',
    'Identifying vital technical keywords to beat automated ATS...',
    'Formulating realistic, customized interview practice questions...',
    'Polishing credentials with professional recruiter vocabulary...'
  ];

  @override
  void dispose() {
    _tipTimer?.cancel();
    super.dispose();
  }

  void _startTipRotation() {
    _tipTimer?.cancel();
    _currentTipIndex = 0;
    _tipTimer = Timer.periodic(const Duration(milliseconds: 3500), (timer) {
      if (mounted) {
        setState(() {
          _currentTipIndex = (_currentTipIndex + 1) % _loadingTips.length;
        });
      }
    });
  }

  void _stopTipRotation() {
    _tipTimer?.cancel();
    _tipTimer = null;
  }

  @override
  Widget build(BuildContext context) {
    final provider = Provider.of<JobsProvider>(context);
    final aiOutput = provider.aiOutput;
    final isAiLoading = provider.isAiLoading;

    if (isAiLoading) {
      if (_tipTimer == null) {
        _startTipRotation();
      }
    } else {
      _stopTipRotation();
    }

    return Scaffold(
      backgroundColor: const Color(0xFF0F172A),
      appBar: AppBar(
        title: Row(
          children: const [
            Icon(Icons.auto_awesome, color: Color(0xFF3B82F6)),
            SizedBox(width: 8),
            Text('AI Career Advisor', style: TextStyle(fontWeight: FontWeight.bold)),
          ],
        ),
        actions: [
          if (aiOutput.isNotEmpty && !isAiLoading) ...[
            IconButton(
              icon: const Icon(Icons.content_copy, color: Color(0xFF94A3B8)),
              tooltip: 'Copy Draft',
              onPressed: () {
                Clipboard.setData(ClipboardData(text: aiOutput));
                ScaffoldMessenger.of(context).showSnackBar(
                  const SnackBar(
                    content: Text('Copied to Clipboard! 📋'),
                    backgroundColor: Color(0xFF10B981),
                  ),
                );
              },
            ),
            IconButton(
              icon: const Icon(Icons.clear, color: Color(0xFFEF4444)),
              tooltip: 'Clear Draft',
              onPressed: () {
                provider.clearAiOutput();
              },
            ),
          ],
        ],
      ),
      body: BoxOrContent(
        isAiLoading: isAiLoading,
        aiOutput: aiOutput,
        tips: _loadingTips,
        tipIndex: _currentTipIndex,
      ),
    );
  }
}

class BoxOrContent extends StatelessWidget {
  final bool isAiLoading;
  final String aiOutput;
  final List<String> tips;
  final int tipIndex;

  const BoxOrContent({
    super.key,
    required this.isAiLoading,
    required this.aiOutput,
    required this.tips,
    required this.tipIndex,
  });

  @override
  Widget build(BuildContext context) {
    if (isAiLoading) {
      return Center(
        child: Padding(
          padding: const EdgeInsets.all(24.0),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              const SizedBox(
                width: 52,
                height: 52,
                child: CircularProgressIndicator(color: Color(0xFF3B82F6), strokeWidth: 4),
              ),
              const SizedBox(height: 24),
              const Text(
                'Drafting with Gemini...',
                style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 18),
              ),
              const SizedBox(height: 16),
              Card(
                color: const Color(0xFF1E293B).withOpacity(0.5),
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(16),
                  side: BorderSide(color: const Color(0xFF334155).withOpacity(0.5)),
                ),
                child: Padding(
                  padding: const EdgeInsets.all(16.0),
                  child: Text(
                    '💡 ${tips[tipIndex]}',
                    style: const TextStyle(color: Color(0xFFE2E8F0), fontSize: 14, height: 1.5),
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
            ],
          ),
        ),
      );
    }

    if (aiOutput.isEmpty) {
      return Center(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(32.0),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: const [
              Icon(Icons.auto_awesome, size: 64, color: Color(0xFF475569)),
              SizedBox(height: 24),
              Text(
                'AI Advisor Standing By',
                style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 20),
                textAlign: TextAlign.center,
              ),
              SizedBox(height: 16),
              Text(
                'Navigate to any job\'s detail page and tap one of our AI tools:\n\n• ✍️ Draft Cover Letter\n• 🎯 Tailor Resume\n• 🎤 Prepare Interview\n\nConfigure your Profile first to let the AI draft details tailored specifically to your background!',
                style: TextStyle(color: Color(0xFF94A3B8), fontSize: 14, height: 1.6),
                textAlign: TextAlign.center,
              ),
            ],
          ),
        ),
      );
    }

    return SingleChildScrollView(
      padding: const EdgeInsets.all(16.0),
      child: Card(
        color: const Color(0xFF1E293B),
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(16),
          side: BorderSide(color: const Color(0xFF334155).withOpacity(0.5)),
        ),
        child: Padding(
          padding: const EdgeInsets.all(16.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Container(
                padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                decoration: BoxDecoration(
                  color: const Color(0xFF3B82F6).withOpacity(0.12),
                  borderRadius: BorderRadius.circular(8),
                ),
                child: Row(
                  mainAxisSize: MainAxisSize.min,
                  children: const [
                    Icon(Icons.auto_awesome, color: Color(0xFF3B82F6), size: 14),
                    SizedBox(width: 6),
                    Text(
                      'Gemini Powered Draft',
                      style: TextStyle(color: Color(0xFF3B82F6), fontWeight: FontWeight.bold, fontSize: 11),
                    ),
                  ],
                ),
              ),
              const SizedBox(height: 16),
              Text(
                aiOutput,
                style: const TextStyle(color: Color(0xFFE2E8F0), fontSize: 14, height: 1.6),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
